src/app/space/create/deployments/apps/deployment-status-icon.component.spec.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ComponentFixture,
+  TestBed
+} from '@angular/core/testing';
+
+import { By } from '@angular/platform-browser';
+import {
+  Component,
+  DebugElement,
+  Input
+} from '@angular/core';
+
+import { Observable } from 'rxjs';
+import { DeploymentStatusIconComponent } from './deployment-status-icon.component';
+
+import { CpuStat } from '../models/cpu-stat';
+
+const ICON_OK = 'pficon-ok';
+const ICON_WARN = 'pficon-warning-triangle-o';
+const ICON_ERR = 'pficon-error-circle-o';
+const MSG_OK = 'Everything is ok.';
+
+describe('DeploymentStatusIconComponent', () => {
+
+  let component: DeploymentStatusIconComponent;
+  let fixture: ComponentFixture<DeploymentStatusIconComponent>;
+  let stat: CpuStat;
+
+  let mockCpuData = Observable.of({ used: 1, total: 5 } as CpuStat) ;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [  ],
+      declarations: [
+        DeploymentStatusIconComponent
+      ]
+    });
+
+    fixture = TestBed.createComponent(DeploymentStatusIconComponent);
+    component = fixture.componentInstance;
+    component.cpuDataStream = mockCpuData;
+  });
+
+  it('should set the button\'s initial value to ok', () => {
+    expect(component.icon === 'pficon-ok');
+    expect(component.icon === 'Everything is ok.');
+  });
+
+  it('should change the button\'s value to warning if capacity changes', () => {
+    expect(component.icon === 'pficon-ok');
+    component.cpuDataStream = Observable.of({ used: 3, total: 5 } as CpuStat);
```

Comment:
A minor enhancement here could be to make cpuDataStream into a `BehaviorSubject`. This would allow you to set the initial value you want, and to trigger changes for each individual test with `cpuDataStream.next(value)`. This would avoid reassigning the entire Observable, which is not exactly how the component will really be used in a live scenario.